src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -153,6 +171,19 @@ public int ResolveProcess(int? pid)
             }
         }
 
+        private async Task<DiagnosticsClient> GetClientAsync(int processId, CancellationToken token)
```

Comment:
At some point our http api surface will have to support Guid instead of pid though right? I'm assuming that would be the next step here.

Reply:
Correct. Wanted to keep that separate since it is not critical for adding the reversed server and this change is already fairly large at this point.